server/src/main/java/org/opensearch/action/search/SearchRequestOperationsCompositeListenerFactory.java

```diff
@@ -69,8 +69,8 @@ public SearchRequestOperationsListener.CompositeListener buildCompositeListener(
         final SearchRequestOperationsListener... perRequestListeners
     ) {
         final List<SearchRequestOperationsListener> searchListenersList = Stream.concat(
-            searchRequestListenersList.stream(),
-            Arrays.stream(perRequestListeners)
+            Arrays.stream(perRequestListeners),
```

Comment:
Please revert this change (the problem will be solved differently)

Reply:
Any suggestions on another way we can solve this issue? @reta

Reply:
There is no issue to solve so far: as we have discussed, using `@InternalApi` is no-go, promoting the API to public will bring the questions:
a) how it is going to be used by plugins (and extensions)
b) what guarantees it will have from the execution perspective

Reply:
The above change was required because of the following discussion : https://github.com/opensearch-project/OpenSearch/pull/14681/files#discussion_r1680221328

Reply:
I remember the discussion, but please refer to the previous comment (https://github.com/opensearch-project/OpenSearch/pull/14832#discussion_r1683438694)

Reply:
> @deshsidd Thanks a lot for following up on the change! However we might be missing one change in this PR: https://github.com/opensearch-project/OpenSearch/pull/14782/files#diff-673f8e993795b1d79af3e0eaee5b2f7254817ce1e18970c74d2c33d9327d281bR72

@ansjcy This change was reverted based on the following comment and thread (reopening the conversation). Let me know your thoughts and what other alternatives we have.

Reply:
Making a change to Node.java to make sure that the core listener runs before the plugin listener for now. Logically, it makes sense to make sure the core listeners are run before the plugin listeners. However, we will do this without making the above change and add the listener directly from Node.java. Discussed with @ansjcy @jainankitk @kkhatua. If needed we can make any additional fixes and refactoring as bug fixes.